src/ape/managers/query.py

```diff
@@ -194,13 +203,18 @@ def query(
                 f" executed in {exec_time} ms (expected: {est_time} ms)"
             )
 
+        skip_caching_plugins = self._get_skip_caching_plugins()
+
         # Update any caches
-        for engine in self.engines.values():
+        for engine_name, engine in self.engines.items():
             if not isinstance(engine, sel_engine.__class__):
-                result, cache_data = tee(result)
-                try:
-                    engine.update_cache(query, cache_data)
-                except QueryEngineError as err:
-                    logger.error(str(err))
+                if engine_name not in skip_caching_plugins:
```

Comment:
refactor nit (feel free to ignore):

```python
if engine_name in skip_caching_plugins:
    logger.debug(f"Skipping cache update for plugin: {engine_name}")
    continue

result, cache_data = tee(result)
try:
    engine.update_cache(query, cache_data)
except QueryEngineError as err:
    logger.error(str(err))
```

is less indenting and less words types